### PR TITLE
Feature/133 migrate search functionality

### LINKED ISSUE #133: Search for titles and books
# Description
Migrate existing search functionality to vuetify.

# DoD:
 - The template in LoanSearch.vue is replaced with Vuetify component.
- Fix lint errors
- Add unit test for searching

# Context
Related #60
Depends on #144

# Discuss with PO
- Clearification of issue (missleading issue title).
- Necessity of tests with PO.

### PR DESCRIPTION
**Pull request by TE4-Daniel-Kull**
## What issue did you implement or fix?
Fixes #133

## Summary
- Renamed LoanSearch to a more generic name, SearchList.
- Migrated bootstrapcode to vuetify

## Type of change
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How can this implementation be improved?
- Add types for items. 

## Context
Relates to #177

### CODE REVIEW

File: frontend/src/components/SearchList.vue [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<template>
```
Comment (TE4-jacob-lundberg) [inline frontend/src/components/SearchList.vue]: Vuetify looks good but make sure to follow our naming standards for files specified in the wiki.

File: frontend/src/components/SearchList.vue [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    </v-card-title>
+    <v-data-table 
+      :headers="headers"
+      :items="items"
+      :search="search"
+    ></v-data-table>
+  </v-card>
+</template>
+
+<script lang="ts">
```
Comment (TE4-jacob-lundberg) [inline frontend/src/components/SearchList.vue]: Might have been branched before the lint was merged but make sure to follow the coding standards defined in the wiki.

File: frontend/src/components/SearchList.vue (hunk 1 above)
Comment (TE4-Daniel-Kull) [reply frontend/src/components/SearchList.vue]: Okidokes! Will fix it right away!

File: frontend/src/router.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import Vue from 'vue';
 import Router from 'vue-router';
 import * as views from '@/views';
+import Test from '@/views/Test.vue';
```
[the remark this answers is not in the document]
Comment (TE4-Daniel-Kull) [reply frontend/src/router.ts]: Good thing you noticed that! It's not supposed to be uploaded, just forgot to remove the import. It's my local test view for testing the components.

File: frontend/src/components/SearchList.vue [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import { Component, Prop, Vue } from "vue-property-decorator";
+
+interface IHeader {
+  text: string,
+  value: string
+}
+
+@Component({})
+export default class SearchList extends Vue {
```
[the remark this answers is not in the document]
Comment (TE4-Daniel-Kull) [reply frontend/src/components/SearchList.vue]: Not currently. Me and Johan are going to use it when we start working on the FindView. :)

File: frontend/src/components/SearchListComponent.vue [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    <v-data-table 
+      :headers="headers"
+      :items="items"
+      :search="search"
+    ></v-data-table>
+  </v-card>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue } from 'vue-property-decorator';
```
Comment (TE4-Mattis-Abrahamsson) [inline frontend/src/components/SearchListComponent.vue]: How about some indentation here?

Comment (TE4-Daniel-Kull) [reply frontend/src/components/SearchListComponent.vue]: The linter didn't fix it, and I'm not sure if it should be. What do you think @TE4-jacob-lundberg ?

Comment (TE4-jacob-lundberg) [reply frontend/src/components/SearchListComponent.vue]: I don't think indentation is necessary here.

**Pull request merged**